tribits/core/package_arch/TribitsReadDepsFilesCreateDepsGraph.cmake

```diff
@@ -616,29 +629,32 @@ macro(tribits_set_dep_packages__handle_undefined_pkg  packageName  depPkg
           " is missing and ${depPkg}_ALLOW_MISSING_EXTERNAL_PACKAGE ="
           " ${${depPkg}_ALLOW_MISSING_EXTERNAL_PACKAGE}!")
       endif()
-      if (requiredOrOptional STREQUAL "REQUIRED")
+      if ("${requiredOrOptional}" STREQUAL "REQUIRED")
```

Comment:
**Question:** Any particular reason for this change? The way you had it before is the canonical way of comparing a variable to some value (though the quoted method works too of course.)

Reply:
This is the deal with macro arguments where I changed from a function() to a macro().  It seems like the only way to use variables or macro arguments in `if()` statements that is robust to all refactorings is to just give up and use `"${<varOrMacroArgName>}"` instead of `<varOrMacroArgName>`. That was the case here (the code is broken if you replace this with `if (requiredOrOptional STREQUAL "REQUIRED")`).  Once I started using `if ( ...  "${<varOrMacroArgName>}"  ...)`, the code behaved much more robust to refactorings.